File: CSharp/HtmlWalker/Tag.Base/ContentScriptTag.cs
using HtmlWalker.Core;
using HtmlWalker.Platform;

namespace HtmlWalker.Tag.Base
{
    /// <summary>
    ///   This class is used to support implementation tag adapters for HTML tags that only have <b>script</b> content
    /// </summary>
    /// <author>Robert R Flores</author>
    public abstract class ContentScriptTag : BaseTag
    {
        /// <summary>
        ///   Instantiates a new content script tag
        /// </summary>
        /// <param name="factory">used to produce objects as per the specific platform</param>
        /// <param name="tag">object housed by this tag adapter</param>
        public ContentScriptTag(WalkerPlatform.WalkerFactory factory, object tag) : base(factory, tag) { }

        /// <summary>
        ///   Implement <see cref="ITag.ContentType"/>
        /// </summary>
        /// <returns><see cref="ContentType.SCRIPT"/></returns>
        public override ContentType ContentType => ContentType.SCRIPT;

        /// <summary>
        ///   Implement <see cref="ITag.TextValue"/>
        /// </summary>
        public override string TextValue { get => Api.GetScript(Tag); set => Api.SetScript(Tag, value); }

        /// <summary>
        ///   Impleents <see cref="ITag.OwnedTags"/>
        /// </summary>
        public override TagList OwnedTags => null;

        /// <summary>
        ///   Impleents <see cref="ITag.OwnedText"/>
        /// </summary>
        public override string OwnedText => TextValue;
    }
}

File: Java/htmlwalker/src/main/java/htmlwalker/tag/base/ContentScriptTag.java
/*
 * 
 */
package htmlwalker.tag.base;

import htmlwalker.platform.*;

/**
 * This class is used to support implementation tag adapters for HTML tags that only have script content.
 * 
 * @author Robert Flores
 */
public abstract class ContentScriptTag extends BaseTag
{
	
	/**
	 * Instantiates a new content script tag.
	 *
	 * @param factory - used to produce objects as per the specific platform
	 * @param tag - object housed by this tag adapter
	 */
	protected ContentScriptTag(WalkerFactory factory, Object tag) { super(factory, tag); }

	/**
	 * Content type.
	 *
	 * @return the content type
	 * @see HtmlWalker.ITag#contentType()
	 */
	@Override
	public ContentType contentType() { return ContentType.SCRIPT; }
	
	/**
	 * Gets the content text.
	 *
	 * @return the content text
	 * @see HtmlWalker.ITag#getTextValue()
	 */
	@Override
	public String getTextValue() { return this.api.getScript(tag()); }

	/**
	 * Sets the content text.
	 *
	 * @param text the new content text
	 * @see HtmlWalker.ITag#setTextValue(java.lang.String)
	 */
	@Override
	public void setTextValue(String text) { this.api.setScript(tag(), text); }

	/**
	 * @see HtmlWalker.ITag#ownedTags()
	 */
	@Override
	public TagList ownedTags() { return null; }
	
	@Override
	public String ownedText() { return getTextValue(); }
}
